libsql/src/hrana/connection.rs

```diff
@@ -1,29 +1,22 @@
-use crate::hrana::pipeline::{
-    ClientMsg, Response, ServerMsg, StreamBatchReq, StreamExecuteReq, StreamRequest,
-    StreamResponse, StreamResponseOk,
-};
+use crate::hrana::pipeline::{BatchStreamReq, ExecuteStreamReq, StreamRequest, StreamResponse};
 use crate::hrana::proto::{Batch, BatchResult, Stmt, StmtResult};
+use crate::hrana::stream::HttpStream;
 use crate::hrana::{HranaError, HttpSend, Result, Statement};
 use crate::util::coerce_url_scheme;
-use std::collections::HashMap;
 use std::sync::atomic::{AtomicI64, AtomicU64, Ordering};
-use std::sync::{Arc, RwLock};
-
-/// Information about the current session: the server-generated cookie
-/// and the URL that should be used for further communication.
-#[derive(Clone, Debug, Default)]
-struct Cookie {
-    baton: Option<String>,
-    base_url: Option<String>,
-}
+use std::sync::Arc;
 
 #[derive(Debug)]
-pub struct HttpConnection<T>(Arc<InnerClient<T>>);
+pub struct HttpConnection<T>(Arc<InnerClient<T>>)
+where
+    T: for<'a> HttpSend<'a>;
```

Comment:
I didn't defined it because I wanted it to be fancy. I defined it this way, because it was the only way I've found to make borrow checker cooperate. If there's a better way I'm open for suggestions.

Reply:
What I mean is that if your struct foo<T> doesn't use the trait inside of the struct def you don't need to bound T by the trait. You can just leave it as T. This will then flow throughout the rest of the code since you no longer need to bound T: Trait for each struct that embeds the struct that contains T. 

Trait bounds should only be set at the impl of where the function is used. The only time you want to bound the struct generic is if you are using an associated type with in the struct.